### DISCUSSION TITLE
Wrong locale date format, but only in booking date field (Firefox / Windows 10)

### DISCUSSION THREAD

**Discussion by Huddeij** [firefly-iii/firefly-iii#8659] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Expecting to have the same localized date format while filling out the form of the booking, but it differs from the headline date info. While the headline show s a correct localized version, the booking date is en/us format:
![image](https://github.com/firefly-iii/firefly-iii/assets/6482096/9bc8993a-5e25-4806-9190-48cb3d795a90)
![image](https://github.com/firefly-iii/firefly-iii/assets/6482096/64137167-6f00-47e5-b90f-72d861654e6c)

### Debug information

No logs of relevance in this situation, I guess. Please adviser me where to look for date format localzation in particular

### Expected behaviour

Showing the same localized date format everywhere in Firefly III

### Steps to reproduce

1. Firefox langauge changed to "en/GB" and "de/DE"
2. According to [this site](https://superuser.com/questions/1751692/firefox-date-format) changed Firefox' config `intl.regional_prefs.use_os_locales` to `true`. Switched "Use your operating system" on and off
3. Restared Firefox
4. Changed localization of WIndows 10 from "en/GB" to "de/DE"
5. restarted FireFly III docker container with `docker compose down && docker compose up -d`

### Additional info

Client: Firefox 123.0.1 on Windows 10
Firefox and Windows are using the language and "en/GB" or "de/DE". Windows uses localzation of "German".
Server: Fedora 39 system with Docker based Firefly III v6.1.10

**Comment by JC5** [comment]
Thanks for reporting this.

Mmm, I've seen it before but not a clue what it could be. The date field uses "datetime-local" and I leave the rest of it to the browser. I'm not sure I can help!

**Comment by Huddeij** [comment]
I feared an answer like this. What is very intriguing is the date format difference between the title row and the actual booking date. What does the top row use, if it is not "datetime-local"?

**Comment by JC5** [comment]
The text is rendered, it's not an input :)

**Reply by hugomferreira** [reply]
I have the exact same symptoms with locale = pt_PT , so far I'm not able to bypass the issue. All my date input fields are with the wrng format (mm - dd - yyyy )

**Comment by matejdro** [comment]
For me, Firefox menu language (not the website language) seem to also affect this:

<img width="835" height="152" alt="image" src="https://github.com/user-attachments/assets/49f0de80-de88-47ef-86a5-1ade06f2ff34" />

**Comment by lethedata** [comment]
Looks like Firefox uses a mix of both os locale and language setting for different things. 

"the language portion of your OS locale matches langauge portion of Firefox locale (for example "en-US" and "en-GB"), then we will use the region portfion of your OS locale in Firefox." [^1]

This field specifically seems to follows whatever the Regional Preferences is showing under  `about:support`.[^2] I confirmed this by only setting the `LC_TIME` env when running.[^3]

What's odd is that @Huddeij seems to be doing everything needed to make it work.[^4] My best guess is that something is triggering firefox to use the default or fallback locale.

[^1]: https://bugzilla.mozilla.org/show_bug.cgi?id=1739712
[^2]: Set by the OS Level: https://firefox-source-docs.mozilla.org/intl/locale.html#regional-preferences
[^3]: `flatpak run --env=LC_TIME=en_GB.utf8 org.mozilla.firefox`
[^4]: I'd be currious what `about:support` shows under Internationalization & Localization
